process/smartContract/process.go

```diff
@@ -175,10 +175,22 @@ func NewSmartContractProcessor(args ArgsNewSmartContractProcessor) (*scProcessor
 	}
 
 	args.EpochNotifier.RegisterNotifyHandler(sc)
+	args.GasSchedule.RegisterNotifyHandler(sc)
 
 	return sc, nil
 }
 
+// GasScheduleChange sets the new gas schedule where it is needed
+func (sc *scProcessor) GasScheduleChange(gasSchedule map[string]map[string]uint64) {
+	apiCosts := gasSchedule[core.ElrondAPICost]
+	if apiCosts == nil {
+		return
+	}
+
+	sc.asyncCallStepCost = apiCosts[core.AsyncCallStepField]
```

Comment:
We need to protect `sc.asyncCallStepCost` and  `sc.asyncCallbackGasLock` with a mutex
Since we are providing the same map pointer this is prone to race conditions. Why not replacing `map[string]map[string]uint64` with and interface that has only some getters with some internal mutex that will protect the node even if it will be used deliberately wrong?

Reply:
no need. there is no possibility to call the from different go routines

Reply:
because if there is an interface it needs to be added like everywhere, vmcommon, vm-util, arwen any change here or there makes it impossible to use.

Reply:
there are no mutexes inside smartContract/process.go at all.

Reply:
protected all the way